### ISSUE TITLE
vnc viewer exited

### ISSUE THREAD

**Issue by hellothere-yan** [ultravnc/UltraVNC#79]
hello，When I connect to my own vnc server, the vnc viewer window exits, but I have no problem using TightVNC viewer.

“vncviewer.exe”(Win32): 已加载“E:\ultra\vncviewer\Debug\vncviewer.exe”。已加载符号。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\ntdll.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\kernel32.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\KernelBase.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\ws2_32.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\rpcrt4.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\WinSxS\x86_microsoft.windows.common-controls_6595b64144ccf1df_6.0.22000.120_none_e541a94fcce8ed6d\comctl32.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\winmm.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\msvcrt.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\user32.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\gdi32.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\win32u.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\win32u.dll”。
“vncviewer.exe”(Win32): 已卸载“C:\Windows\SysWOW64\win32u.dll”
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\comdlg32.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\gdi32full.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\ucrtbase.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\msvcp_win.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\combase.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\SHCore.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\shlwapi.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\shell32.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\advapi32.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\sechost.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\imm32.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\WinSxS\x86_microsoft.windows.gdiplus_6595b64144ccf1df_1.1.22000.1098_none_94058a64d9e96c8c\GdiPlus.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\windows.storage.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\WinTypes.dll”。
Started and Winsock (v 2) initialised
bufsize expanded to 4352
Registered connection with app
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\riched32.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\riched20.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\msls31.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\usp10.dll”。
Saving to E:\ultra\vncviewer\Debug\options.vnc
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\d3d9.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\dwmapi.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\kernel.appcore.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\DXCore.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\uxtheme.dll”。
DPI 96
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\msctf.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\bcryptprimitives.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\TextInputFramework.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\oleaut32.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\CoreMessaging.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\CoreUIComponents.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\cryptbase.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\ole32.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\clbcatq.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\SogouTSF.ime”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\msimg32.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\version.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\SogouPY.ime”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\oleacc.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\winhttp.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\ntmarta.dll”。
“vncviewer.exe”(Win32): 已加载“D:\My_soft\输入法\SogouInput\Components\PicFace\1.1.0.1881\PicFace.dll”。
“vncviewer.exe”(Win32): 已加载“D:\My_soft\输入法\SogouInput\11.6.0.5419\Resource.dll”。
“vncviewer.exe”(Win32): 已卸载“D:\My_soft\输入法\SogouInput\11.6.0.5419\Resource.dll”
“vncviewer.exe”(Win32): 已加载“D:\My_soft\输入法\SogouInput\11.6.0.5419\Resource.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\profapi.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\psapi.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\TextShaping.dll”。
线程 0x988c 已退出，返回值为 0 (0x0)。
“vncviewer.exe”(Win32): 已加载“C:\Program Files (x86)\Sangfor\SSL\SangforPWEx\SangforUDProtectEx.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\crypt32.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Program Files (x86)\Sangfor\SSL\SangforPWEx\sangforvpnlibcrypto-1_1.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\wintrust.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\wtsapi32.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\msasn1.dll”。
线程 0x88bc 已退出，返回值为 0 (0x0)。
“vncviewer.exe”(Win32): 已卸载“C:\Windows\SysWOW64\SogouTSF.ime”
Reset keyboard for first use
Flush dead key gives: 1 character(s): 0x0020 ( )
“vncviewer.exe”(Win32): 已加载“C:\Program Files (x86)\Sangfor\SSL\ClientComponent4\SangforTcp.dll”。模块已生成，不包含符号。
“vncviewer.exe”(Win32): 已卸载“C:\Program Files (x86)\Sangfor\SSL\ClientComponent4\SangforTcp.dll”
“vncviewer.exe”(Win32): 已加载“C:\Program Files (x86)\Sangfor\SSL\ClientComponent4\SangforTcp.dll”。模块已生成，不包含符号。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\msvcp60.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\urlmon.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\IPHLPAPI.DLL”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\mfc42.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\dbghelp.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\iertutil.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\wininet.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\srvcli.dll”。
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\netutils.dll”。
CSProxy refcount 161
Loading E:\ultra\vncviewer\Debug\vncviewer.exe...
WSPStartup ===> E:\ultra\vncviewer\Debug\vncviewer.exe使用链式SPI“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\mswsock.dll”。
[WSPConnect] Socket ip 10.17.136.197:5900Connected to 10.17.136.197 port 5900
DSMPlugin not enabled
RFB server supports protocol version 3.8
Connected to RFB server, using protocol version 3.8
“vncviewer.exe”(Win32): 已加载“C:\Windows\SysWOW64\SogouTSF.ime”。
VNC authentication succeeded
Read a 7-byte string
Desktop name "BMC VNC "
Geometry 1024 x 768 depth 32
Screen work area is 1920 x 1032
Reset keyboard for first use
Flush dead key gives: 1 character(s): 0x0020 ( )
Clipboard changed
Ignore Clipboard while initializing!
No next window in chain; WM_DRAWCLIPBOARD will not be passed
SetClipboardViewer to 0x00901a36; next is 0x00000000. Last error 0x00000000Screen work area is 1920 x 1032
DPI changed, SizeWindow
Screen work area is 1920 x 1032
Update-processing thread started
Screen work area is 1920 x 1032
Screen work area is 1920 x 1032
线程 0x5e40 已退出，返回值为 0 (0x0)。
线程 0x48dc 已退出，返回值为 0 (0x0)。
Receiving cursor shape update, cursor 20x20
bufsize expanded to 22656
zlibbufsize expanded to 65792
zlibbufsize expanded to 65792
zlibbufsize expanded to 65792
zlibbufsize expanded to 65792
zlibbufsize expanded to 65792
zlibbufsize expanded to 65792
zlibbufsize expanded to 65792
zlibbufsize expanded to 65792
zlibbufsize expanded to 65792
zlibbufsize expanded to 65792
zlibbufsize expanded to 65792
DPI changed, new=144 old=96
DPI changed, SizeWindow
Screen work area is 1920 x 1008

**Comment by hellothere-yan** [comment]
hello ,anybody here?

**Comment by RudiDeVos** [comment]
Yes, but with just a log trace is hard to tell what is going on.
The only thing i see is that the server is switching resolution a few times

**Comment by hellothere-yan** [comment]
![Snipaste_2023-03-29_19-27-10](https://user-images.githubusercontent.com/45117047/228521830-db317294-b2fc-4bd8-a144-e36c8afe6ae1.jpg)
I noticed that my mouse keeps flickering too

**Comment by hellothere-yan** [comment]
Thank you for your reply, I am using [libvncserver](https://github.com/LibVNC/libvncserver), the encoding method I use is tight, I want to adapt to ultra VNC,but the viewer quits after working for a while.

**Comment by hellothere-yan** [comment]
i found that the resolution sent by my server during initialization is consistent with the resolution of the video frame I actually send. it is 1027*768

**Comment by hellothere-yan** [comment]
here is my wireshark data.
![11](https://user-images.githubusercontent.com/45117047/229013699-ce71b5da-cde9-41c5-94eb-15fe15f74249.jpg)

**Comment by hellothere-yan** [comment]
From the packet  data, it seems that the data sent by the viewer to the server is wrong.I do not know why ultra VNC viewer send  data like this.

**Comment by hellothere-yan** [comment]
do you need the packet data ?

**Comment by Neustradamus** [comment]
@RudiDeVos: Please look @hellothere-yan comments :)

**Comment by RudiDeVos** [comment]
I need to be able to run the viewer in debug ( Visual studio) while he is connected to that server, trace and log just doesn't provide enough info.
Can you provide a server or put some server on the net i can use to connect to ?

**Comment by hellothere-yan** [comment]
@RudiDeVos I'm afraid not. My server is a BMC KVM of a linux server.IT is hard to put it on server.Is there any other way?thank you

**Comment by hellothere-yan** [comment]
my server is based on obmc-ikvm[https://github.com/openbmc/obmc-ikvm](url)，can you load it?

**Comment by hellothere-yan** [comment]
I noticed that my mouse keeps flickering ，What could be the cause of this?

**Comment by hellothere-yan** [comment]
Is the mouse blinking the cause of the vnc viewer exiting?

**Comment by hellothere-yan** [comment]
I use tight encoding, and there is no problem with tigger VNC。

**Comment by hellothere-yan** [comment]
I have single-step debugging in visual studio, but I still haven't found the reason why the viewer exits. How can I make the viewer stop before the program exits. Several times while stepping through, my computer became unresponsive.

**Comment by hellothere-yan** [comment]
According to what you said, I added breakpoints at all exit functions, but when vncviewer exited, the program did not stay at the breakpoints.it's weird
![break](https://user-images.githubusercontent.com/45117047/230254610-0e5bf739-f232-45b1-af8a-6eb9c8893dd6.jpg)
![break1](https://user-images.githubusercontent.com/45117047/230255117-829590b0-e656-47f0-9833-1ffca7689ef9.jpg)

**Comment by hellothere-yan** [comment]
yes ,According to your suggestion, I added a breakpoint to the exit function and found that the program was broken here
![break](https://user-images.githubusercontent.com/45117047/230263649-69333bc6-4817-4968-b67d-925519a0dc6e.jpg)

**Comment by hellothere-yan** [comment]
Following your advice, I found where the program exits
![break](https://user-images.githubusercontent.com/45117047/230331616-db89adc6-4b25-4ad9-a825-aa78db07b670.jpg)

**Comment by hellothere-yan** [comment]
This seems to be looking for the beginning(0xffd8) and end of the picture(0xffd9), and then exits if it can't find it. However, through my wireshark packet capture data analysis, my picture frame data has a beginning(0xffd8) and a end(0xffd9).the following is my last packet,
![frame](https://user-images.githubusercontent.com/45117047/230377201-d820c361-35bb-4402-8b5d-08321934aa84.jpg)

**Comment by RudiDeVos** [comment]
Looks like the tight decoder crash when decompessing a jpeg block.
As you are using libvncserver, can you try  with an another encoder.

We are using libjpeg-turbo and libvncServer also seems to use it, still some decompression error.

Even windows to windows this would be a hell to debug as it happen in the jpeg library

void ClientConnection::DecompressJpegRect(int x, int y, int w, int h) abd verify if xywh is actual part of the desktop.

**Comment by RudiDeVos** [comment]
1.4.2.0 have update libs, including libjpeg-turbo... you never know

**Comment by hellothere-yan** [comment]
Yes, I updated to 1.4.2.0, but there will still be exits, and it is the same as before.
![Snipaste_2023-04-17_10-07-19](https://user-images.githubusercontent.com/45117047/232361326-180d8e41-a8e9-4db4-ac64-993c8b4f0e6c.jpg)
![break](https://user-images.githubusercontent.com/45117047/232361336-6f74959d-025c-4e96-a194-282aca260abd.jpg)

**Comment by Neustradamus** [comment]
@hellothere-yan: Have you tested the latest dev build or previous stable?
- https://forum.uvnc.com/viewtopic.php?t=37855

**Comment by hellothere-yan** [comment]
> @hellothere-yan: Have you tested the latest dev build or previous stable?
> 
> * https://forum.uvnc.com/viewtopic.php?t=37855

yes,I tested version 1.4.3.3 and unfortunately still have the problem, it looks like the problem is the same

**Comment by RudiDeVos** [comment]
We don't know if it's caused by the server or viewer
If i can't repeat the crash, i can not fix it.
Sorry

**Comment by Neustradamus** [comment]
@hellothere-yan: What is your situation?

There is a 1.4.3.6 and a 1.5.0.0-dev, can you try and explain if the problem is always here?

**Comment by Neustradamus** [comment]
@hellothere-yan: It is possible to have a comeback from you?

Thanks in advance.

**Comment by hellothere-yan** [comment]
> @hellothere-yan: It is possible to have a comeback from you?
> 
> Thanks in advance.

The 1.5.0.0-dev has the same error. viewer run for a while before exiting.

**Comment by Neustradamus** [comment]
@hellothere-yan: Can you try 1.5.0.1-dev?
- https://forum.uvnc.com/viewtopic.php?t=37976

**Comment by Neustradamus** [comment]
@hellothere-yan: Can you try 1.5.0.3-dev?

**Comment by hellothere-yan** [comment]
> @hellothere-yan: Can you try 1.5.0.3-dev?

hello ，I couldn't find 1.5.0.3-dev, but I updated the latest code and found that it was the same error. I was using the tight encoding method and the mouse was still flashing wildly. It quit shortly after。
tigerVNC and tightVNC  has no errors occurred.what can i do for you to slove this problem? thanks.

**Comment by Neustradamus** [comment]
@hellothere-yan: It is cited in topic here: https://forum.uvnc.com/viewtopic.php?t=37976

cc: @RudiDeVos.

**Comment by Neustradamus** [comment]
@hellothere-yan: Can you test with 1.5.0.5?
- https://forum.uvnc.com/viewtopic.php?t=38037

Can you add logs for @RudiDeVos?

**Comment by hellothere-yan** [comment]
> [@hellothere-yan](https://github.com/hellothere-yan): Can you test with 1.5.0.5?
> 
> * https://forum.uvnc.com/viewtopic.php?t=38037
> 
> Can you add logs for [@RudiDeVos](https://github.com/RudiDeVos)?

I tested version 1.5.0.5, but the problem still occurs. Which logs should I provide? My VNCserver is obmc-ikvm of openbmc.
https://github.com/[openbmc/obmc-ikvm](https://github.com/openbmc/obmc-ikvm)

**Comment by Neustradamus** [comment]
@hellothere-yan: Some people have tested UltraVNC without problem:
- https://github.com/openbmc/obmc-ikvm/issues/6

**Comment by hellothere-yan** [comment]
> [@hellothere-yan](https://github.com/hellothere-yan): Some people have tested UltraVNC without problem:
> 
> * [Using VNC Viewer to connect openbmc/obmc-ikvm#6](https://github.com/openbmc/obmc-ikvm/issues/6)

I tested it again using tight encoding. When the OS is shut down, there is no problem. When the OS is turned on, ultra VNC will exit. My BMC chip is ast2600. I am not sure if it has something to do with the BMC chip, because the image frame is provided by ast2600. I used tightvnc viewer, tigervnc viewer and no VNC on the same device, and there was no problem.

**Comment by hellothere-yan** [comment]
> [@hellothere-yan](https://github.com/hellothere-yan): Some people have tested UltraVNC without problem:
> 
> * [Using VNC Viewer to connect openbmc/obmc-ikvm#6](https://github.com/openbmc/obmc-ikvm/issues/6)

this is my log.




Jan 14 11:51:36 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:36 Got connection from client 10.2.46.172
Jan 14 11:51:36 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:36   0 other clients
Jan 14 11:51:36 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:36 Client Protocol Version 3.8
Jan 14 11:51:36 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:36 Protocol version sent 3.8, using 3.8
Jan 14 11:51:36 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:36 rfbProcessClientSecurityType: executing handler for type 1
Jan 14 11:51:36 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:36 rfbProcessClientSecurityType: returning securityResult for client rfb version >= 3.8
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 Pixel format for client 10.2.46.172:
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37   32 bpp, depth 32, little endian
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37   true colour: max r 255 g 255 b 255, shift r 0 g 8 b 16
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 no translation needed
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0x0000001A)
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0x0000001D)
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0x0000001B)
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0x00000019)
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0x00000013)
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0x00000012)
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0x0000000A)
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 rfbProcessClientNormalMessage: ignoring unsupported encoding type zlibhex
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 Using compression level 6 for client 10.2.46.172
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 Enabling full-color cursor updates for client 10.2.46.172
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 Enabling cursor position updates for client 10.2.46.172
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 Using image quality level 8 for client 10.2.46.172
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 Using JPEG subsampling 0, Q92 for client 10.2.46.172
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFF000B)
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 Enabling LastRect protocol extension for client 10.2.46.172
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 Enabling NewFBSize protocol extension for client 10.2.46.172
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 Enabling ExtDesktopSize protocol extension for client 10.2.46.172
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFF8000)
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFF8001)
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFF8004)
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFF8002)
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFF8003)
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFF8005)
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 Enabling ExtendedClipboard extension for client 10.2.46.172
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 Using tight encoding for client 10.2.46.172
Jan 14 11:51:37 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:37 Sending rfbEncodingExtDesktopSize for size (1024x768)
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41 rfbProcessClientNormalMessage: read: Connection reset by peer
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41 Client 10.2.46.172 gone
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41 Statistics             events    Transmit/ RawEquiv ( saved)
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41  ServerCutText       :      1 |        16/       16 (  0.0%)
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41  FramebufferUpdate   :      1 |         0/        0 (  0.0%)
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41  LastRect            :    104 |      1248/     1248 (  0.0%)
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41  tight               :    104 |   2878661/327156960 ( 99.1%)
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41  RichCursor          :      1 |      1684/     1684 (  0.0%)
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41  ExtendedDesktopSize :      1 |        32/       32 (  0.0%)
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41  TOTALS              :    212 |   2881641/327159940 ( 99.1%)
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41 Statistics             events    Received/ RawEquiv ( saved)
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41  ClientCutText       :      1 |        28/       28 (  0.0%)
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41  PointerEvent        :      5 |        30/       30 (  0.0%)
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41  FramebufferUpdate   :    106 |      1060/     1060 (  0.0%)
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41  SetEncodings        :      1 |       136/      136 (  0.0%)
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41  SetPixelFormat      :      1 |        20/       20 (  0.0%)
Jan 14 11:51:41 e8611a096589 obmc-ikvm[1256]: 14/01/2024 11:51:41  TOTALS              :    114 |      1274/     1274 (  0.0%)
^A

**Comment by Neustradamus** [comment]
@hellothere-yan: Can you try a recent UltraVNC version > 1.6?
Please add a log, thanks in advance.

**Comment by hellothere-yan** [comment]
> [@hellothere-yan](https://github.com/hellothere-yan): Can you try a recent UltraVNC version > 1.6? Please add a log, thanks in advance.

hello ,I tried version 1.6 .1.0,but it failed.

![Image](https://github.com/user-attachments/assets/3bbf500d-1934-4cc2-89cd-8942b0f67d27)

same error as before.
Can you contact me(yanzt08@163.com) for remote debugging?  what log should i add?

**Comment by hellothere-yan** [comment]
> The server seems to be balking at `Sending rfbEncodingExtDesktopSize for size (1024x768)`
> 
> Try checking `GNOME R.D.` option in `Display` tab in connection's options.

Whether this GNOME R.D. option is enabled or not, the error is the same.

![Image](https://github.com/user-attachments/assets/e27e4194-a546-4dca-8af9-eb222ad2d4d0)

**Comment by hellothere-yan** [comment]
> [@hellothere-yan](https://github.com/hellothere-yan): Can you try UltraVNC 1.6.4.0 build?
> 
> * https://forum.uvnc.com/viewforum.php?f=56

hello,I test the 1.6.4.0 version, not good.

![Image](https://github.com/user-attachments/assets/63bff13d-a002-4ef0-9cd6-adf59a03fa2f)

**Comment by Neustradamus** [comment]
@hellothere-yan: Can you send the full log for @RudiDeVos?

**Comment by hellothere-yan** [comment]
I printed the first 64 bytes of data and found that there was indeed an error marker information. I consulted the reference material and found that libjpeg strictly follows the JPEG standard. If there is an error in the image frame data, it will cause the exit function to be called, causing the viewer to exit. From this point of view, it is very likely that there is a problem with the image data returned by the aspeed chip. But I think there should be a fault tolerance mechanism.
Here is wrong frame date.
FF D8 FF E0 00 10 4A 46 49 46 00 01 01 01 00 60 
00 60 00 00 FF FE 00 0F 05 2D 01 3C 00 00 FF 6C 
02 19 E8 46 00 FF DB 00 43 00 14 0D 0F 11 0F 0C 
14 11 10 11 16 15 14 17 1E 32 20 1E 1B 1B 1E 3D 
JPEG FATAL ERROR: Unsupported marker type 0x6c
JPEG decoding failed!

**Comment by wqweto** [comment]
This is a C++ codebase so instead of set/longjmp you can try `throw ErrorException("Fatal JPEG error")`.

If you plan on sending a PR keep in mind that `jpeg_create_decompress` is called in two places so both need the fix.

**Comment by hellothere-yan** [comment]
> This is a C++ codebase so instead of set/longjmp you can try `throw ErrorException("Fatal JPEG error")`.
> 
> If you plan on sending a PR keep in mind that `jpeg_create_decompress` is called in two places so both need the fix.

According to the previous breakpoint information, the error should be in jpeg_read_header, not jpeg_create_decompress. I will submit a PR for C++ style exception handling.

**Comment by Neustradamus** [comment]
@hellothere-yan: Thanks for your PR :)
- https://github.com/ultravnc/UltraVNC/pull/285

@RudiDeVos has recently merged this one!
